app/src/main/java/com/example/mystore/ui/login/LoginScreen.kt

```diff
@@ -51,126 +70,158 @@ fun LoginScreen(navigateToProductDetailsScreen: () -> Unit = {}) {
     var isButtonEnabled by remember { mutableStateOf(true) }
-    val mContext = LocalContext.current
     val coroutineScope = rememberCoroutineScope()
-    var loginViewModel: LoginViewModel = hiltViewModel()
-    val errorMessage = "Your email or password are incorrect"
+    val snackbarHostState = remember { SnackbarHostState() }
+    val loginViewModel: LoginViewModel = hiltViewModel()
+    var errorMessage: String? by remember { mutableStateOf(null) }
     val isValidEmail: (email: String) -> Boolean = { email ->
-        android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()
+        EMAIL_ADDRESS.matcher(email).matches()
     }
     val validateFields: (email: String, password: String) -> Boolean = { email, password ->
         email.isNotBlank() &&
                 isValidEmail(email) &&
                 password.isOverSevenCharacters() &&
                 password.hasDigits()
     }
-    Column(
-        modifier = Modifier
-            .padding(30.dp)
-            .padding(top = 180.dp)
-    ) {
-        Text(
-            modifier = Modifier.padding(bottom = 20.dp),
-            text = "Log in",
-            fontSize = MaterialTheme.typography.headlineLarge.fontSize,
-            textAlign = TextAlign.Center,
-            fontWeight = FontWeight.Bold
-        )
-        OutlinedTextField(
+
+    LaunchedEffect(loginViewModel.response?.errorBody()) {
+        val errorBody = loginViewModel.response?.errorBody()?.string()
+        if (errorBody != null) {
+            val jObjError = JSONObject(errorBody)
+            errorMessage =
+                jObjError.getJSONObject("error")
+                    .getString("message")
+            snackbarHostState.showSnackbar(errorMessage.toString())
+        } else {
+            snackbarHostState.showSnackbar("Unknown error occurred")
+        }
+    }
+    Scaffold(
+        snackbarHost = { SnackbarHost(hostState = snackbarHostState) }
+    ) { paddingValues ->
+        Box(
             modifier = Modifier
-                .fillMaxWidth()
-                .padding(bottom = 20.dp),
-            value = emailField,
-            onValueChange = {
-                emailField = it
-                isEmailValid = isValidEmail(it)
-            },
-            isError = !isEmailValid && emailField.isNotEmpty(),
-            placeholder = { Text(text = "Email") },
-            label = { Text("Email") },
-            singleLine = true,
-            keyboardOptions = KeyboardOptions(
-                keyboardType = KeyboardType.Email,
-                imeAction = ImeAction.Next
-            ),
-            supportingText = {
-                if (!isEmailValid && emailField.isNotEmpty()) {
-                    Text(text = "Invalid email", color = Color.Red)
-                }
+                .fillMaxSize()
+                .padding(paddingValues)
+        ) {
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(top = 150.dp),
```

Comment:
No random numbers -- try to export them and reuse them. There are some others, look for those in the code and refactor them (no need to spam you with comments about it).